src/core/unit.c

```diff
@@ -343,20 +343,11 @@ int unit_choose_id(Unit *u, const char *name) {
         if (!s)
                 return -ENOENT;
 
-        /* Determine the new instance from the new id */
-        r = unit_name_to_instance(name, &i);
```

Comment:
this doesn't look right actually, the instance can differ between aliases. Specifically socket units that have Accept=yes synthesize service units which have a number as instance first, already when binding the socket, since they need to calculate an selinux label. Later on, when the first connection comes in, this socket unit is then updated with an instance generated from socket metadata. The unit then has two instance names.

Reply:
I was working on this patchset to fix the CI failure that @mrc0mmand pasted, and before I saw your comment I reworked the code to not do that... I get why this was done, but it seems very brittle and breaks too many assumptions.